Text/Mustache/Parser.hs

```diff
@@ -152,12 +152,30 @@ pClosingTag key = do
 {-# INLINE pClosingTag #-}
 
 pKey :: Parser Key
-pKey = (fmap Key . lexeme . label "key") (implicit <|> other)
+pKey =
+  (fmap Key . lexeme . label "key") $
+    implicit <|> do
+      end <- gets closingDel
+
+      -- pKey is not allowed to contain letters from closingDel.
+      --
+      -- This is an undocumented behaviour. We are employing the same strategy as
+      -- found in the package `mustache`.
+      let f x = x `notElem` ('.' : '}' : T.unpack end)
+          lbl = "characters different from those found in \"" ++ (T.unpack end) ++ "\""
+
+      other f lbl
   where
     implicit = [] <$ char '.'
-    other = sepBy1 (takeWhile1P (Just lbl) f) (char '.')
-    lbl = "alphanumeric char or '-' or '_'"
-    f x = isAlphaNum x || x == '-' || x == '_'
+
+    other :: (Char -> Bool) -> String -> Parser [Text]
```

Comment:
Let's stick to the expression with `septBy1`, it reads much clearer. Instead, do `fmap` over its result and strip tailing whitespace in the last segment of the key, *as well as leading whitespace in the first segment*. See what this demo does:

http://mustache.github.io/#demo

Reply:
Okay, I can change it as you suggest. Btw, I considered stripping the first segment as well but it made no difference, the first segment is already stripped for some reason.